### PR TITLE
Fix `--help` with ecosystem extensions

### PR DESCRIPTION
**Pull request by cd-work**
Closes https://github.com/phylum-dev/cli/issues/989.

### CODE REVIEW

**Review by kylewillmon — CHANGES_REQUESTED** [review]
The `npm` extension now outputs the help twice, which is only marginally better than the previous behavior.

<details>
<summary>`phylum npm install --help` output</summary>

```
> phylum npm install --help
[phylum] Updating lockfile…
Install a package

Usage:
npm install [<package-spec> ...]

Options:
[-S|--save|--no-save|--save-prod|--save-dev|--save-optional|--save-peer|--save-bundle]
[-E|--save-exact] [-g|--global]
[--install-strategy <hoisted|nested|shallow|linked>] [--legacy-bundling]
[--global-style] [--omit <dev|optional|peer> [--omit <dev|optional|peer> ...]]
[--strict-peer-deps] [--no-package-lock] [--foreground-scripts]
[--ignore-scripts] [--no-audit] [--no-bin-links] [--no-fund] [--dry-run]
[-w|--workspace <workspace-name> [-w|--workspace <workspace-name> ...]]
[-ws|--workspaces] [--include-workspace-root] [--install-links]

aliases: add, i, in, ins, inst, insta, instal, isnt, isnta, isntal, isntall

Run "npm help install" for more info
[phylum] No lockfile created.

[phylum] Installing without build scripts…
Install a package

Usage:
npm install [<package-spec> ...]

Options:
[-S|--save|--no-save|--save-prod|--save-dev|--save-optional|--save-peer|--save-bundle]
[-E|--save-exact] [-g|--global]
[--install-strategy <hoisted|nested|shallow|linked>] [--legacy-bundling]
[--global-style] [--omit <dev|optional|peer> [--omit <dev|optional|peer> ...]]
[--strict-peer-deps] [--no-package-lock] [--foreground-scripts]
[--ignore-scripts] [--no-audit] [--no-bin-links] [--no-fund] [--dry-run]
[-w|--workspace <workspace-name> [-w|--workspace <workspace-name> ...]]
[-ws|--workspaces] [--include-workspace-root] [--install-links]

aliases: add, i, in, ins, inst, insta, instal, isnt, isnta, isntal, isntall

Run "npm help install" for more info
[phylum] Packages installed successfully.

[phylum] Running build scripts inside sandbox…

up to date, audited 1 package in 29ms

found 0 vulnerabilities
[phylum] Packages built successfully.
```
</details>

The `pip` extension also runs the command twice, but it doesn't have the same issue because the first call (the dry-run) is run with output piped, so it doesn't print to the terminal.

File: extensions/npm/main.ts [hunk 1]
```diff
@@ -176,7 +176,13 @@ async function checkDryRun(subcommand: string, args: string[]) {
     await abort(status.code);
   }
 
-  const lockfile = await PhylumApi.parseLockfile("./package-lock.json", "npm");
+  let lockfile;
+  try {
+    lockfile = await PhylumApi.parseLockfile("./package-lock.json", "npm");
+  } catch (_e) {
+    console.warn(`[${yellow("phylum")}] No lockfile created.\n`);
+    return;
```
Comment (kylewillmon) [inline extensions/npm/main.ts:184]: You can probably exit here instead of return.

```suggestion
    await abort(0);
```

Comment (cd-work) [reply extensions/npm/main.ts:184]: I don't think we should, since we did not run the proper command yet at this point.

**Comment by cd-work** [comment]
> The npm extension now outputs the help twice, which is only marginally better than the previous behavior.

I'm aware of this but I think considering what we do it's the best solution and it works generically regardless if the flag is called help or anything else. I'd rather avoid a fragile implementation that tries to parse the arguments itself.

**Review by kylewillmon — APPROVED** [review]

**Review by maxrake — APPROVED** [review]
Local testing was performed to verify functionality. The only limitation found with `--help` is when it is the first/only flag provided (and there are no sub-commands specified). This affects the `pip`, `poetry`, and `yarn` extensions, but not the `npm` extension:

```console
~/dev/delme/npm_init  1  3 is 📦 v1.0.0 via  v19.9.0 took 26s
❯ ~/dev/phylum/localdev/cli/target/debug/phylum npm --help
npm <command>

Usage:

npm install        install all the dependencies in your project
npm install <foo>  add the <foo> dependency to your project
npm test           run this project's tests
npm run <foo>      run the script named <foo>
npm <command> -h   quick help on <command>
npm -l             display usage info for all commands
npm help <term>    search for help on <term>
npm help npm       more involved overview

All commands:

    access, adduser, audit, bugs, cache, ci, completion,
    config, dedupe, deprecate, diff, dist-tag, docs, doctor,
    edit, exec, explain, explore, find-dupes, fund, get, help,
    help-search, hook, init, install, install-ci-test,
    install-test, link, ll, login, logout, ls, org, outdated,
    owner, pack, ping, pkg, prefix, profile, prune, publish,
    query, rebuild, repo, restart, root, run-script, search,
    set, shrinkwrap, star, stars, start, stop, team, test,
    token, uninstall, unpublish, unstar, update, version, view,
    whoami

Specify configs in the ini-formatted file:
    /Users/maxrake/.npmrc
or on the command line via: npm <command> --key=value

More configuration info: npm help config
Configuration fields: npm help 7 config

npm@9.6.3 /opt/homebrew/lib/node_modules/npm
```

```console
../cli  14  24 on  extension_help [?] via 🦕 via 🐍 v3.11.3 via 🦀 v1.68.2
❯ target/debug/phylum pip --help
[phylum] This extension does not support arguments before the first subcommand. Please open an issue if "--help" is not an argument.
```

```console
~/dev/delme/npm_init  1  3 is 📦 v1.0.0 via  v19.9.0
❯ ~/dev/phylum/localdev/cli/target/debug/phylum yarn --help
[phylum] This extension does not support arguments before the first subcommand. Please open an issue if "--help" is not an argument.
```

```console
../phylum-ci  9  22 on  main [?] via 🐳 desktop-linux is 📦 v0.28.1 via 🐍 v3.11.3
❯ ~/dev/phylum/localdev/cli/target/debug/phylum poetry --help
[phylum] This extension does not support arguments before the first subcommand. Please open an issue if "--help" is not an argument.
```

Marking approved anyway since the issue this PR links to fix is #989, which only addresses the subcommand `--help` usage.

It would still be nice to address the top-level `--help` usage (perhaps in a different PR) because our documentation still suggests users alias these package manager commands and if they do, they will almost certainly be surprised when basic help output is not available. I'll make a new issue for this.

**Pull request merged**